Sources/OfflineEntitlements/CustomerInfo+OfflineEntitlements.swift

```diff
@@ -0,0 +1,71 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  CustomerInfo+OfflineEntitlements.swift
+//
+//  Created by Nacho Soto on 3/21/23.
+
+import Foundation
+
+@available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
+extension CustomerInfo {
+
+    convenience init(
+        from purchasedSK2Products: [PurchasedSK2Product],
+        mapping: ProductEntitlementMapping,
+        userID: String,
+        sandboxEnvironmentDetector: SandboxEnvironmentDetector = BundleSandboxEnvironmentDetector.default
+    ) {
+        let subscriber = CustomerInfoResponse.Subscriber(
+            originalAppUserId: userID,
+            managementUrl: Self.defaultManagementURL,
+            originalApplicationVersion: SystemInfo.buildVersion,
+            originalPurchaseDate: Date(),
+            firstSeen: Date(),
+            subscriptions: purchasedSK2Products
+                .dictionaryAllowingDuplicateKeys { $0.productIdentifier }
+                .mapValues { $0.subscription },
+            nonSubscriptions: [:],
+            entitlements: Self.createEntitlements(with: purchasedSK2Products, mapping: mapping)
+        )
+
+        let content: CustomerInfoResponse = .init(
+            subscriber: subscriber,
+            requestDate: Date(),
+            rawData: (try? subscriber.asDictionary()) ?? [:]
+        )
+
+        self.init(
+            response: content,
+            entitlementVerification: Self.verification,
+            sandboxEnvironmentDetector: sandboxEnvironmentDetector
+        )
+    }
+
+    private static func createEntitlements(
+        with products: [PurchasedSK2Product],
+        mapping: ProductEntitlementMapping
+    ) -> [String: CustomerInfoResponse.Entitlement] {
+        var result: [String: CustomerInfoResponse.Entitlement] = .init(minimumCapacity: products.count)
+
+        for product in products {
+            for entitlement in mapping.entitlements(for: product.productIdentifier) {
+                result[entitlement] = product.entitlement
```

Comment:
Hmm if the user purchased multiple products that grant the same entitlement (probably an uncommon situation), we would keep the latest one in the array... Should we keep the one with the latest expiration date instead?

Reply:
Oh good point! I'll write a test and improve this.

Reply:
Oh wait, but they all have the same expiration since we're not saving the `StoreKit.Transaction.expiration`, simply setting it to tomorrow.

Reply:
Ahh true.... Then it's probably ok to just pick the latest

Reply:
thoughts: 
- we should use the actual transaction expiration, unless there's a good reason not to that I'm missing
- we should check how khepri selects entitlements and mimic

Reply:
Done: a44cbffe23dc2cd518cb58e10426db83e0f22725